### PR TITLE
Refactor into pkg

### PR DESCRIPTION
**Pull request by Smirl**
# Summary

- Move the http server and handlers into `pkg/infrabin`
- Move `internal/helpers` to `pkg/helpers`
- Add `HTTPServer` struct (can be used to have loggers, etc in the future such as https://github.com/stefanprodan/podinfo/blob/master/pkg/api/server.go#L72)
- Add graceful shutdown
- Allows us to add a grpc server in a new file later

### CODE REVIEW

File: cmd/go-infrabin/main.go [hunk 1]
```diff
@@ -1,176 +1,31 @@
...
+
+	// run admin server in background
+	admin := infrabin.NewAdminServer()
+	go admin.ListenAndServe()
+
+	// wait for SIGINT
 	<-finish
+
+	admin.Shutdown()
+	server.Shutdown()
```
Comment (Smirl) [inline cmd/go-infrabin/main.go:30]: This does mean that it shuts down one then the other.
Need better `context` foo.

File: pkg/infrabin/http.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
+package infrabin
```
Comment (Smirl) [inline pkg/infrabin/http.go:1]: This is just a copy paste from main.

File: pkg/infrabin/server.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	log.Printf("Starting %s server on %s", s.Name, s.Server.Addr)
+	if err := s.Server.ListenAndServe(); err != http.ErrServerClosed {
+		log.Fatal("HTTP server crashed", err)
+	}
+}
+
+
+func (s *HTTPServer) Shutdown() {
+	log.Printf("Shutting down %s server with 15s graceful shutdown", s.Name)
+	ctx, cancel := context.WithTimeout(context.Background(), 15 * time.Second)
```
Comment (Smirl) [inline pkg/infrabin/server.go:28]: Will wait up to 15 seconds or exit immediately
Followed this https://github.com/gorilla/mux#graceful-shutdown

Comment (maruina) [reply pkg/infrabin/server.go:28]: Awesome, thanks

File: cmd/go-infrabin/main.go (hunk 1 above)
Comment (maruina) [reply cmd/go-infrabin/main.go:30]: Not sure what you mean here. What we should do?

**Comment by maruina** [comment]
The reason why it had the previous structure it's because I was following https://github.com/golang-standards/project-layout. In particular when they say about `internals`
```
Private application and library code. This is the code you don't want others importing in their applications or libraries
```
I'm not sure what are the best practices or implication about moving the helpers into package. What do you think?

File: cmd/go-infrabin/main.go (hunk 1 above)
Comment (Smirl) [reply cmd/go-infrabin/main.go:30]: I would say it is ok for now.

**Pull request merged**